Sources/PackageDescription/Package.swift

```diff
@@ -169,6 +169,23 @@ extension Package {
         dict["dependencies"] = .array(dependencies.map({ $0.toJSON() }))
         dict["exclude"] = .array(exclude.map({ .string($0) }))
         dict["targets"] = .array(targets.map({ $0.toJSON() }))
+
+        // Find duplicate targets
+        var uniqueTargetNames = Set<String>()
+        var duplicateTargets  = Array<Target>()
+        for target in targets {
```

Comment:
We move this logic to `PackageBuilder`, as that is more appropriate. The errors here are supposed to be usually parsing (or similar) errors.